assets/dev/js/frontend/handlers/image-carousel.js

```diff
@@ -2,63 +2,132 @@ class ImageCarouselHandler extends elementorModules.frontend.handlers.Base {
-					settings: {
-						slidesToShow: +elementSettings.slides_to_show_tablet || defaultLGDevicesSlidesCount,
-						slidesToScroll: +elementSettings.slides_to_scroll_tablet || defaultLGDevicesSlidesCount,
-					},
-				},
-				{
-					breakpoint: breakpoints.md,
-					settings: {
-						slidesToShow: +elementSettings.slides_to_show_mobile || 1,
-						slidesToScroll: +elementSettings.slides_to_scroll_mobile || 1,
-					},
-				},
-			],
 		};
 
-		if ( isSingleSlide ) {
-			slickOptions.fade = 'fade' === elementSettings.effect;
-		} else {
-			slickOptions.slidesToScroll = +elementSettings.slides_to_scroll || defaultLGDevicesSlidesCount;
+		swiperOptions.breakpoints = {};
+
+		swiperOptions.breakpoints[ elementorBreakpoints.md ] = {
+			slidesPerView: +elementSettings.slides_to_show_mobile || 1,
+			slidesPerGroup: +elementSettings.slides_to_scroll_mobile || 1,
+		};
+
+		swiperOptions.breakpoints[ elementorBreakpoints.lg ] = {
+			slidesPerView: +elementSettings.slides_to_show_tablet || defaultLGDevicesSlidesCount,
+			slidesPerGroup: +elementSettings.slides_to_scroll_tablet || defaultLGDevicesSlidesCount,
+		};
+
+		if ( ! this.isEdit && elementSettings.autoplay ) {
+			swiperOptions.autoplay = {
+				delay: elementSettings.autoplay_speed,
+				disableOnInteraction: !! elementSettings.pause_on_interaction,
+			};
 		}
 
-		this.elements.$carousel.slick( slickOptions );
+		if ( true === swiperOptions.loop ) {
+			swiperOptions.loopedSlides = this.getSlidesCount();
+		}
+
+		if ( elementSettings.image_spacing_custom ) {
+			swiperOptions.spaceBetween = elementSettings.image_spacing_custom.size;
+		}
+
+		const showArrows = 'arrows' === elementSettings.navigation || 'both' === elementSettings.navigation;
+		const showDots = 'dots' === elementSettings.navigation || 'both' === elementSettings.navigation;
+
+		if ( showArrows ) {
+			swiperOptions.navigation = {
+				prevEl: '.elementor-swiper-button-prev',
+				nextEl: '.elementor-swiper-button-next',
+			};
+		}
+
+		if ( showDots ) {
+			swiperOptions.pagination = {
+				el: '.swiper-pagination',
+				type: 'bullets',
+				clickable: true,
+			};
+		}
+
+		return swiperOptions;
+	}
+
+	getSpaceBetween() {
+		var propertyName = 'image_spacing_custom';
+
+		return this.getElementSettings( propertyName ).size || 0;
+	}
+
+	updateSpaceBetween( swiper ) {
+		var newSpaceBetween = this.getSpaceBetween();
+
+		swiper.params.spaceBetween = newSpaceBetween;
+
+		swiper.update();
+	}
+
+	onInit( ...args ) {
+		super.onInit( ...args );
+
+		if ( ! this.elements.$carousel.length ) {
+			return;
+		}
+
+		this.swiper = new Swiper( this.elements.$carousel, this.getSwiperSettings() );
+	}
+
+	onElementChange( propertyName ) {
+		if ( 0 === propertyName.indexOf( 'width' ) ) {
+			this.swiper.update();
+		}
+
+		if ( 0 === propertyName.indexOf( 'image_spacing_custom' ) ) {
+			this.updateSpaceBetween( this.swiper, propertyName );
+		}
+	}
+
+	onEditSettingsChange( propertyName ) {
```

Comment:
Why is it needed?

Reply:
the entire onEditSettingsChange method?
Because of the slideToLoop method call - it is needed in order for the slide to change in the editor's front-end when it is selected in the panel